contrib/cirrus/test.sh

```diff
@@ -67,7 +67,14 @@ else
             showrun make validate
             ;;
         unit)
-            showrun make test-unit
+
+            if [[ -n "$CIRRUS_PR" ]]; then
+                showrun make test-unit
+            else
+		# If not running on a PR then run unit tests
+		# with appropriate `-race` flags.
+                showrun RACEFLAGS=true make test-unit
+            fi
```

Comment:
Then here pass `RACEFLAGS=-race`, and I would avoid duplicating the full command, the if/else should just focus on the changed part. IMO this would be more readable

```sh
local race=
if [[ -z "$CIRRUS_PR" ]]; then
	# If not running on a PR then run unit tests
	# with appropriate `-race` flags.
	race="-race"
fi
showrun RACEFLAGS=$race make test-unit
```

Also indentation looks wrong